src/Promise.ts

```diff
@@ -473,8 +473,8 @@ export default class Promise<T> implements Thenable<T> {
 	/**
 	 * Adds a callback to the promise to be invoked when the asynchronous operation completes successfully.
 	 */
-	then<U>(onFulfilled?: (value?: T) => (U | Thenable<U>), onRejected?: (reason?: Error) => void): Promise<U>;
-	then<U>(onFulfilled?: (value?: T) => (U | Thenable<U>), onRejected?: (reason?: Error) => (U | Thenable<U>)): Promise<U> {
+	then<U>(onFulfilled?: ((value?: T) => (U | Thenable<U>)) | null, onRejected?: (reason?: Error) => void): Promise<U>;
```

Comment:
Why `| null` here instead of `| undefined`?  From a usability perspective, maybe it needs to be `| null | undefined` as I think developers would expect to pass either.